python/cudf/dataframe/categorical.py

```diff
@@ -86,7 +86,8 @@ def __init__(self, **kwargs):
         """
         categories = kwargs.pop('categories')
         ordered = kwargs.pop('ordered')
-        kwargs.update({'dtype': pd.core.dtypes.dtypes.CategoricalDtype()})
+        dtype = pd.core.dtypes.dtypes.CategoricalDtype(categories, ordered)
```

Comment:
This may cause issues where we check for dtype equivalence elsewhere.

Reply:
I don't know enough about how dtypes are compared, but I can totally see this causing perf issues if CategoricalDtype compares its category values.

This was necessary because the `columnops.column_empty()` function is only passed the dtype instance, not the entire cudf column. Since `column_empty()` is used to create an all-null column that's filled in later, the empty column returned doesn't have the same categories as the column it's cloning. This line here https://github.com/rapidsai/cudf/pull/2012/files#diff-b303068aeb1cfec667f31ad7b3348c41R130 is where I pluck the categories from the dtype if it's a categorical.

Reply:
Doesn't seem to be breaking anything so should be good to go!